### ISSUE TITLE
why do you multiply mul(1.0-1.0/s[1]).mul(n) ?

### ISSUE THREAD

**Issue by jawaechan** [jiecaoyu/XNOR-Net-PyTorch#121]
Hi, 
    In your gradient update, you multiplied mul(1.0-1.0/s[1]).mul(n) to the gradient, which does not appear in your note.pdf. Could you explain why you multiplied these two terms?

Thanks

**Comment by Chenyaoyi1998** [comment]
I have the same problem. 
The authors answered why they multiplied 1e+9 under another question, but not why mul(1.0-1.0/s[1]).mul(n)，
But I have a feeling it's all for a similar purpose, to minimise the effects of weight decay, and the difference with 1e+9 is that this can be correlated with s and n.
